DigitalLearningSolutions.Data/Services/NotificationPreferencesDataService.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using System.Collections.Generic;
+    using System.Data;
+    using Dapper;
+    using DigitalLearningSolutions.Data.Models;
+
+    public interface INotificationPreferencesDataService
+    {
+        IEnumerable<NotificationPreference> GetNotificationPreferencesForAdmin(int? adminId);
+        IEnumerable<NotificationPreference> GetNotificationPreferencesForDelegate(int? delegateId);
+    }
+
+    public class NotificationPreferencesDataService : INotificationPreferencesDataService
+    {
+        private readonly IDbConnection connection;
+
+        public NotificationPreferencesDataService(IDbConnection connection)
+        {
+            this.connection = connection;
+        }
+
+        public IEnumerable<NotificationPreference> GetNotificationPreferencesForAdmin(int? adminId)
+        {
+            if (!adminId.HasValue)
+            {
+                return new List<NotificationPreference>();
+            }
+
+            return connection.Query<NotificationPreference>(
+                @"DECLARE @CentreAdmin BIT, @IsCentreManager BIT, @ContentManager BIT, @ContentCreator BIT, @Supervisor BIT, @Trainer BIT
+
+                    SELECT @CentreAdmin = CentreAdmin, @IsCentreManager = IsCentreManager, @ContentManager = ContentManager, @ContentCreator = ContentCreator, @Supervisor = Supervisor, @Trainer = Trainer
+	                    FROM AdminUsers
+	                    WHERE AdminID = @adminId
+
+                    DECLARE @Whitespace VARCHAR(3)

```

Comment:
The notification names have unhelpful whitespace and new lines at the end, which caused some problems when setting up the tests. Stripping them to make things easier downstream and leaves the question of how to display the name entirely in the hands of the View.

Reply:
Double checked with Kevin and he said that it's probably just some bad handling form the old system so they're fine to remove! :)